From a7fb3b863c3574470c45fdf6084dcc40931ca017 Mon Sep 17 00:00:00 2001
From: Francis Laniel <flaniel@linux.microsoft.com>
Date: Thu, 7 Apr 2022 18:30:23 +0100
Subject: [PATCH] Link against libabseil-cpp which grpc uses.

This patch was taken from:
55c96b61f7b9 ("sysdig: make extra linking against abseil conditional") [1]
from hhoffstaette/portage.

[1] https://github.com/hhoffstaette/portage/blob/55c96b61f7b91e4d91bed6723e86c00dd91f8d16/dev-util/sysdig/files/0.27.1-grpc-absl-sync.patch

Signed-off-by: Francis Laniel <flaniel@linux.microsoft.com>
---
 CMakeLists.txt                    | 1 +
 userspace/libsinsp/CMakeLists.txt | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1f34f1d6..c0354b29 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -552,6 +552,7 @@ if(NOT WIN32 AND NOT APPLE)
 			else()
 				message(FATAL_ERROR "Couldn't find system grpc")
 			endif()
+			find_library(ABSL_SYNC_LIB NAMES absl_synchronization)
 			find_program(GRPC_CPP_PLUGIN grpc_cpp_plugin)
 			if(NOT GRPC_CPP_PLUGIN)
 				message(FATAL_ERROR "System grpc_cpp_plugin not found")
diff --git a/userspace/libsinsp/CMakeLists.txt b/userspace/libsinsp/CMakeLists.txt
index d72f9115..d1b7ea02 100644
--- a/userspace/libsinsp/CMakeLists.txt
+++ b/userspace/libsinsp/CMakeLists.txt
@@ -214,6 +214,10 @@ if(NOT WIN32)
 				"${JQ_LIB}"
 				"${B64_LIB}")
 
+			if(ABSL_SYNC_LIB)
+				target_link_libraries(sinsp "${ABSL_SYNC_LIB}")
+			endif()
+
 			if(NOT MUSL_OPTIMIZED_BUILD)
 			target_link_libraries(sinsp
 				rt
-- 
2.25.1

